Refactor: Auto-derive W100 external sensors and mode switch from aqara_w100 + simplify inputs and fix HVAC sync logic

maxipallotto opened the pull request:
Key changes
- Removed manual inputs
w100_external_temperature, w100_external_humidity, and w100_thermostat_mode are no longer user-selected.
- Dynamic entity derivation
External temperature, humidity, and thermostat-mode switch entities are now automatically generated from the selected aqara_w100 (e.g., climate.temp_hum_office → number.temp_hum_office_external_temperature, etc.).
- Simplified and more robust blueprint
Fewer inputs, fewer configuration errors, and more consistent behavior across all sync paths.

maxipallotto committed:
Remove W100 external temperature and humidity inputs 

Removed external temperature and humidity input from the configuration.
Replace corresponding entities by auto generated ones from W100 Climate input.

Add | float to the template so it renders numeric value

maxipallotto committed:
Fix YAML formatting for entity_id in actions

maxipallotto committed:
Refactor w100 thermostat mode handling

Removed w100 thermostat mode input and updated entity_id references to use a template for dynamic resolution.

maxipallotto committed:
Refactor thermostat mode condition to use template

maxipallotto committed:
Refactor thermostat mode conditions in YAML

maxipallotto committed:
Rollback thermostat mode condition to use template

nurtext commented:
@KipK Will this be merged soon? Seems like it's an important PR to me...

KipK commented:
Thanks @maxipallotto 
How does it handles people who renamed their entities path ? 
Like translated, or different convention names.

KipK commented:
I'd prefer we keep the manual entities selection to avoid the edge case above, but keep your fixes.

nurtext commented:
@KipK Seems like a good compromise to me.

KipK commented:
@maxipallotto could you separate the fixes in your PR ? 

Other changes are clearly too specific as I could see. Not only the entity names can be changed, but device too. As example mine is not even called "aqara_w100" in my ha , and then the blueprint is all broken :)

One you could do is prepopulate the entities if they're found, but let separated fields for each entities

maxipallotto committed:
Enhance fan mode synchronization logic

Refactor fan mode handling with improved variable normalization and conditions.

maxipallotto committed:
Update value_template formatting in w100-blueprint.yaml

Refactor value_template to use block style and improve readability.

Felix-Chaos commented:
Just got a look into this
is this at a working state?
and if yes how do i install this correctly so that the w100 actually displays more than current temp and humidity?

maxipallotto commented:
@KipK Sorry for the late reply, you know end of year.
No fixes in my fork. Just simplified your blueprint  by auto generated ones from "Aqara W100 Climate Controller" input
And YES you can rename the device or entities and will still work (It's also renamed in my HA setup)

KipK commented:
hi @maxipallotto , i had to revert this [PR](https://github.com/KipK/aqara-w100-z2m-converter/pull/7) as it was causing some issues. Haven't checked yet what's wrong.
Can you resync your PR to main?

Thanks

maxipallotto committed:
Merge branch 'KipK:main' into main

maxipallotto commented:
Hi @KipK done. Also incorporated fan fixes from main branch

KipK commented:
Thanks for this, cool change
